src/MoBi.Core/Commands/RenameObjectBaseCommand.cs

```diff
@@ -48,13 +51,23 @@ protected virtual void RenameObjectBase(IMoBiContext context)
       {
          OldName = _objectBase.Name;
          _objectBase.Name = _newName;
+
          if (_objectBase.IsAnImplementationOf<IBuildingBlock>())
          {
-            var renameBuildingBlockTask = context.Resolve<IRenameBuildingBlockTask>();
-            renameBuildingBlockTask.RenameInSimulationUsingTemplateBuildingBlock(_objectBase.DowncastTo<IBuildingBlock>());
+            doWithRenameInSimulationTask(task => task.RenameInSimulationUsingTemplateBuildingBlock(OldName, _objectBase.DowncastTo<IBuildingBlock>()), context);
+         }
+         else if (_objectBase.IsAnImplementationOf<Module>())
+         {
+            doWithRenameInSimulationTask(task => task.RenameInSimulationUsingTemplateModule(OldName, _objectBase.DowncastTo<Module>()), context);
          }
       }
 
+      private void doWithRenameInSimulationTask(Action<IRenameInSimulationTask> action, IMoBiContext context)
```

Comment:
I am not sure I get this. What about you instantiate this tast in the calling method and then just call the method instead of creating a method with a delegate. Can't see the benefit of this indirectionhere

Reply:
It's just about removing duplicates of context.Resolve. I don't want to resolve the task if it's not going to be used and I didn't want to just duplicate that line. But, it's probably a lot more straightforward with

```
if(something)
  resolveThisThing
else if (something else)
  resolveThisThing
```

Reply:
sure. But the task is the same so put it before the if and it's a clear call then
Plus use switch on type so that you don't downcast etc